### DISCUSSION TITLE
Как работать с рефлексией и атрибутами в C#

### DISCUSSION THREAD

**Discussion by pingvin1308** [cleannetcode/LessonMonitor#22] [ASP.NET Core]
Для выполнения домашней работы, вам необходимо открыть и сделать форк основного репозитория по ссылке:

По выполнению работы, вам необходимо создать пулл реквест из вашей ветки с выполненной домашкой в основной репозиторий.

**Почитать:**
https://metanit.com/sharp/tutorial/14.1.php
https://docs.microsoft.com/ru-ru/dotnet/framework/reflection-and-codedom/reflection

**Сделать:**
Добавить метод в контроллер, который возвращает метаданные классов в определенном неймспейсе:
 - Создать директорию (неймспейс) и придумать ей название
 - Прочитать все классы с данного неймспейса
 - Вывести метаданные:  
    - Название класса
    - Свойств класса (название свойства, описание свойства, тип свойства)

Создать свой атрибут для проверки валидности модели и применить его в методе контроллера.
К примеру таким образом можно проверить входной параметр метода.

**Проверить:**
провести ревью кода любого чужого пулл реквеста относящегося к данной домашке.

**Comment by Misha902** [comment]
Это нужно продолжать прошлую ветку с домашкой или создать новую?

**Reply by eniluck** [reply]
https://youtu.be/Pv8OXPXP9Wg?t=9158

**Comment by Jenausmax** [comment]
Роман, а что понимается под описанием свойства? ///Summery ? или что-то еще?

**Reply by pingvin1308** [reply]
Описание свойства это просто атрибут Description. Было бы прикольно, если бы вы реализовали данный атрибут самостоятельно, а затем применили его в коде)
